### PR TITLE
Add support for domain socket on m1 mac

### PR DESCRIPTION
**Pull request by jcleezer**
Epoll domain sockets arent supported on M1 mac. In these cases Kqueue should be used. This changes the factories to check if epoll or kqueue is available when creating the domain socket channel.

### CODE REVIEW

File: r2-netty/src/main/java/com/linkedin/r2/netty/client/http2/Http2ChannelPoolFactory.java [hunk 1]
```diff
@@ -116,8 +119,19 @@ public Http2ChannelPoolFactory(
...
+    } else {
+      if (Epoll.isAvailable()) {
+        channelClass = EpollDomainSocketChannel.class;
+      } else if (KQueue.isAvailable()){
+        channelClass = KQueueDomainSocketChannel.class;
+      } else {
+        throw new IllegalStateException("Neither Epoll or Kqueue domain socket transport available");
+      }
+    }
+    Bootstrap bootstrap = new Bootstrap().channel(channelClass);
```
Comment (TylerHorth) [inline r2-netty/src/main/java/com/linkedin/r2/netty/client/http2/Http2ChannelPoolFactory.java:134]: Can we extract the above logic into a util and deduplicate the code?

Comment (jcleezer) [reply r2-netty/src/main/java/com/linkedin/r2/netty/client/http2/Http2ChannelPoolFactory.java]: Yeah that was a bit of a smell. Moved to a common location. Thanks!

**Review by TylerHorth — APPROVED** [review]
LGTM

**Review by zackthehuman — APPROVED** [review]

**Review by zackthehuman — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,11 @@ and what APIs have changed, if applicable.
 
 ## [Unreleased]
 
+## [29.49.5] - 2024-01-11
+
+## [29.4.5] - 2024-01-11
```
Comment (zackthehuman) [inline CHANGELOG.md:19]: Remove these two lines.

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,11 @@ and what APIs have changed, if applicable.
...
+- Added KQueue support for domain sockets.
+-
```
Comment (zackthehuman) [inline CHANGELOG.md:21]: Replace this with a blank line.

**Review by zackthehuman — CHANGES_REQUESTED** [review]

**Review by zackthehuman — APPROVED** [review]
Thanks for the fix!

**Pull request merged**